### ISSUE TITLE
No packets captured when using wireless devices

### ISSUE THREAD

**Issue by odedshimon** [chmorgan/sharppcap#301]
Hi,

I don't know if it is an expected behavior or maybe a misuse (done by me) at the library:
I'm successfully capturing packets from Ethernet devices, but when I'm selecting wireless devices I get no packets (the OnPacketArrival event is not raising at any point).
I ask this because the conditions are the same in both cases (same code, same computer).

Did I miss something? Do you have any idea how to proceed with the investigation of the problem?

This where I'm using the live capturing at my project: https://github.com/odedshimon/BruteShark/blob/master/BruteShark/PcapProcessor/Sniffer.cs
This is the issue opened by the project users that which made me think there maybe a problem with using wireless devices: https://github.com/odedshimon/BruteShark/issues/99

**Comment by kayoub5** [comment]
My experience with wireless interfaces is that they behave identically to ethernet devices by default (unless changed to monitor mode)

Wireless interfaces do not log packets unless the wireless device is connected to a network and is actively exchanging packets with the router. 

Some details that could identify or solve the problem:
* use latest version of npcap driver 
* use latest version of sharppcap

Note: some interfaces/drivers do not support Promiscuous Mode, old versions of sharppcap throw an error in this case, new versions of sharppcap allow ignoring that setting promiscuous mode failed.

**Comment by odedshimon** [comment]
OK I figured it out (thanks to the lead!).
I have tested it again using latest versions (SharpPcap 6.0.0 + Npcap 1.50), still no change compared to the previous test.

Then I used the [capturing example] (https://github.com/chmorgan/sharppcap/blob/master/Examples/Example3.BasicCap/Program.cs) on the same environment and luckily it worked.

At this point I realized that the problem is probably with my code (along with your lead, that from your experience wireless interfaces behave identically to ethernet devices). I simultaneously debugged my code along with the example above code example and found an elusive bug at my code.

Thanks!
